commit 1bc3a0b41c0c119cd049160a87b09ea95a4532a0
Author: Luca Beltrame <lbeltrame@kde.org>
Date:   Wed Jun 14 15:57:54 2017 +0200

    Revert "When building menu hyerarchies, parent menus to their containers"
    
    This causes crashes in KPart using applications. Like the other reverted
    fix, the issue this was trying to solve is handled by
    https://codereview.qt-project.org/#/c/195704/
    
    This reverts commit b5af0296cb2f060d9427551e07f470c4d7c39b85.
    
    CCMAIL: montel@kde.org

diff --git src/kxmlguifactory_p.cpp src/kxmlguifactory_p.cpp
index eb28cb2..5481f93 100644
--- src/kxmlguifactory_p.cpp
+++ src/kxmlguifactory_p.cpp
@@ -25,7 +25,6 @@
 
 #include <QWidget>
 #include <QDebug>
-#include <QMenu>
 
 #include "debug.h"
 #include <assert.h>
@@ -593,9 +592,6 @@ bool BuildHelper::processActionElement(const QDomElement &e, int idx)
 
     parentNode->container->insertAction(before, action);
 
-    if (action->menu() && !action->menu()->parentWidget()) {
-        action->menu()->setParent(parentNode->container, action->menu()->windowFlags());
-    }
     // save a reference to the plugged action, in order to properly unplug it afterwards.
     containerClient->actions.append(action);
 
